sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -112,7 +112,12 @@ class SimpleFunctionRegistry extends FunctionRegistry {
         throw new AnalysisException(s"undefined function $name")
       }
     }
-    func(children)
+    try {
+      func(children)
```

Comment:
Although this is the simplest fix, I think ideally it's the `FunctionBuilder`'s responsibility to check invalid number of arguments, like the built-in function's `FunctionBuilder` does.